libp2p-xtra/src/endpoint.rs

```diff
@@ -381,30 +382,48 @@ impl Endpoint {
         let this = ctx.address().expect("we are alive");
         let listen_address = msg.0.clone();
 
-        self.listen_addresses.insert(listen_address.clone()); // FIXME: This address could be a "catch-all" like "0.0.0.0" which actually results in
-                                                              // listening on multiple interfaces.
```

Comment:
If possible, can we keep this one here and remove the line in the last commit with the rationale that we want to wait until the call to `listen_on` succeeds before we add it?

Reply:
I'm afraid I don't follow. are you talking about the  the comment or the whole 2 lines? and remove the whole solution from last commit?

Reply:
I think this commit should not touch these two lines and instead we can move the insert from here to what we have in the last commit!